From 19de297a4344b21a72772f49bc4a3c9e26e3810f Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Fri, 22 Jun 2018 15:04:29 -0700
Subject: [PATCH] MANTA-3785 waferlock allowing destroyed VMs MANTA-3786
 waferlock doesn't deal with old VMs with no "ips" property on NICs Reviewed
 by: Trent Mick <trent.mick@joyent.com> Reviewed by: Cody Peter Mello
 <cody.mello@joyent.com> Approved by: Cody Peter Mello <cody.mello@joyent.com>

---
 lib/sapi.js | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/lib/sapi.js b/lib/sapi.js
index c9e4ef2..b3e254a 100644
--- a/lib/sapi.js
+++ b/lib/sapi.js
@@ -249,10 +249,21 @@ SapiPoller.prototype.state_runq_do = function (S) {
 		mod_assert.arrayOfObject(objs);
 
 		objs.forEach(function (vm) {
-			var ips = [];
+			if (vm.state === 'destroyed' || vm.destroyed)
+				return;
+			if (vm.state === 'failed')
+				return;
+			if (!vm.nics || !Array.isArray(vm.nics))
+				return;
 
+			var ips = [];
 			vm.nics.forEach(function (nic) {
-				nic.ips.forEach(function (ip) {
+				if (typeof (nic) !== 'object')
+					return;
+				var ipa = nic.ips;
+				if (ipa === undefined)
+					ipa = [nic.ip];
+				ipa.forEach(function (ip) {
 					var parts = ip.split('/');
 					ips.push(parts[0]);
 				});
-- 
2.21.0

